### PR TITLE
Adds support for serde(flatten)

### PR DESCRIPTION
**Pull request by dunnock**
This change adds support for #[serde(flatten)] item attribute for Apiv2Schema derive macro. This allows to reuse structs for building flat JSON queries and responses like in example below:
```
    #[derive(Deserialize, Serialize, Apiv2Schema)]
    struct Paging {
        /// Starting image number
        offset: i32,
        /// Total images found
        total: i32,
        /// Page size
        size: i32,
    };

    #[derive(Serialize, Apiv2Schema)]
    struct Image {
        data: String,
        id: Uuid,
        time: chrono_dev::DateTime<chrono_dev::Utc>,
    }

    /// Images response with paging information embedded
    #[derive(Serialize, Apiv2Schema)]
    struct Images {
        data: Vec<Image>,
        #[serde(flatten)]
        paging: Paging,
    }
```

will turn into
```
                        "Images": {
                          "properties": {
                            "data": {
                              "items": {
                                "properties": {
                                  "data": {
                                    "type": "string"
                                  },
                                  "id": {
                                    "format": "uuid",
                                    "type": "string"
                                  },
                                  "time": {
                                    "format": "date-time",
                                    "type": "string"
                                  }
                                },
                                "required": [
                                  "data",
                                  "id",
                                  "time"
                                ]
                              },
                              "type": "array"
                            },
                            "offset": {
                              "description": "Starting image number",
                              "format": "int32",
                              "type": "integer"
                            },
                            "size": {
                              "description": "Page size",
                              "format": "int32",
                              "type": "integer"
                            },
                            "total": {
                              "description": "Total images found",
                              "format": "int32",
                              "type": "integer"
                            }
                          },
                          "required": [
                            "data",
                            "paging"
                          ]
                        }
```

### CODE REVIEW

File: macros/src/actix.rs [hunk 1]
```diff
@@ -1057,3 +1064,37 @@ impl SerdeProps {
         props
     }
 }
+
+/// Supported flattening of embedded struct (https://serde.rs/variant-attrs.html).
+struct SerdeFlatten;
+
+impl SerdeFlatten {
+    /// Traverses the field attributes and returns the renamed value from the first matching
+    /// `#[serde(rename = "...")]` pattern.
```
Comment (wafflespeanut) [inline macros/src/actix.rs:1073]: `serde(flatten)` :smile:

File: macros/src/actix.rs [hunk 2]
```diff
@@ -1057,3 +1064,37 @@ impl SerdeProps {
...
+    fn from_field_attrs(field_attrs: &[Attribute]) -> bool {
```
Comment (wafflespeanut) [inline macros/src/actix.rs:1074]: Let's rename this to `exists` because it returns a boolean.

File: macros/src/actix.rs [hunk 3]
```diff
@@ -845,20 +845,27 @@ fn handle_field_struct(
...
+        let flatten = SerdeFlatten::from_field_attrs(&field.attrs);
+
         let (ty_ref, is_required) = get_field_type(&field);
 
         let docs = extract_documentation(&field.attrs);
         let docs = docs.trim();
 
-        let mut gen = quote!(
-            {
+        let mut gen = if !flatten {
```
Comment (wafflespeanut) [inline macros/src/actix.rs:855]: Once we rename the function (as mentioned below), this could be `if SerdeFlatten::exists(&field.attrs) { ... }`

**Review by wafflespeanut — CHANGES_REQUESTED** [review]
Great work! Just a few nits...

File: macros/src/actix.rs (hunk 1 above)
Comment (dunnock) [reply macros/src/actix.rs]: Fixed

File: macros/src/actix.rs (hunk 2 above)
Comment (dunnock) [reply macros/src/actix.rs]: good point!

File: macros/src/actix.rs (hunk 3 above)
Comment (dunnock) [reply macros/src/actix.rs]: Cool, this looks better now. Just applied all suggestions

**Review by wafflespeanut — APPROVED** [review]
Thank you!

**Pull request merged**